cmd/mdatagen/internal/samplereceiver/generated_component_telemetry_test.go

```diff
@@ -0,0 +1,72 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package samplereceiver
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+)
+
+type componentTestTelemetry struct {
+	reader        *sdkmetric.ManualReader
+	meterProvider *sdkmetric.MeterProvider
+}
+
+func (tt *componentTestTelemetry) NewCreateSettings() receiver.CreateSettings {
+	settings := receivertest.NewNopCreateSettings()
+	settings.MeterProvider = tt.meterProvider
+	settings.ID = component.NewID(component.MustNewType("sample"))
+
+	return settings
+}
+
+func setupTestTelemetry() componentTestTelemetry {
+	reader := sdkmetric.NewManualReader()
+	return componentTestTelemetry{
+		reader:        reader,
+		meterProvider: sdkmetric.NewMeterProvider(sdkmetric.WithReader(reader)),
```

Comment:
I think either the metric provider or the manual reader needs to be shutdown after the test. Perhaps store the mp somewhere, and provide a shutdown function?

Reply:
Definitely would be good to have a `Shutdown(context)` on the componentTestTelemetry struct.

Reply:
Shutdown added and used in the samplereceiver test